build.gradle.kts

```diff
@@ -33,17 +33,68 @@ nexusPublishing {
 
 description = "OpenTelemetry instrumentations for Java"
 
-// total of 4 partitions (see modulo 4 below)
-var testPartition = (project.findProperty("testPartition") as String?)?.toInt()
-if (testPartition != null) {
-  var testPartitionCounter = 0
+if (project.findProperty("skipTests") as String? == "true") {
   subprojects {
-    // relying on predictable ordering of subprojects
-    // (see https://docs.gradle.org/current/dsl/org.gradle.api.Project.html#N14CB4)
-    // since we are splitting these tasks across different github action jobs
-    val enabled = testPartitionCounter++ % 4 == testPartition
     tasks.withType<Test>().configureEach {
-      this.enabled = enabled
+      enabled = false
+    }
+  }
+}
+
+tasks {
+  val listTestPartitions by registering {
+    group = "Help"
+    description = "List test tasks in given partition"
+
+    // total of 4 partitions (see modulo 4 below)
+    var testPartition = (project.findProperty("testPartition") as String?)?.toInt()
+    if (testPartition == null) {
+      throw GradleException("Test partition must be specified")
+    } else if (testPartition < 0 || testPartition >= 4) {
+      throw GradleException("Invalid test partition")
+    }
+
+    val partitionTasks = ArrayList<Test>()
+    var testPartitionCounter = 0
+    subprojects {
+      // relying on predictable ordering of subprojects
+      // (see https://docs.gradle.org/current/dsl/org.gradle.api.Project.html#N14CB4)
+      // since we are splitting these tasks across different github action jobs
+      val enabled = testPartitionCounter++ % 4 == testPartition
+      if (enabled) {
+        tasks.withType<Test>().configureEach {
```

Comment:
Just to make sure: that includes the gradle test suites, right?

Reply:
It does. For example in https://github.com/open-telemetry/opentelemetry-java-instrumentation/actions/runs/4530824015/jobs/7980235634?pr=8142 when you open `Test` step you can see the command line which includes the test tasks passed to gradle. It includes `:instrumentation:spring:spring-jms:spring-jms-2.0:javaagent:testReceiveSpansDisabled` which is a test suite.
Definitely one of the risks with this is that it is not straightforward and there is a chance that some test might be never run if this logic turns out to be faulty.